src/client/Microsoft.Identity.Client/Internal/Requests/ManagedIdentityAuthRequest.cs

```diff
@@ -38,87 +38,117 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             }
 
             AuthenticationResult authResult = null;
-            MsalAccessTokenCacheItem cachedAccessTokenItem = null;
-            var logger = AuthenticationRequestParameters.RequestContext.Logger;
+
+            ILoggerAdapter logger = AuthenticationRequestParameters.RequestContext.Logger;
             CacheRefreshReason cacheInfoTelemetry = CacheRefreshReason.NotApplicable;
 
-            if (!_managedIdentityParameters.ForceRefresh)
+            //skip checking cache for force refresh
+            if (_managedIdentityParameters.ForceRefresh)
             {
-                cachedAccessTokenItem = await CacheManager.FindAccessTokenAsync().ConfigureAwait(false);
+                cacheInfoTelemetry = CacheRefreshReason.ForceRefreshOrClaims;
```

Comment:
`ApiEvent.CacheInfo` needs to be set here instead.

nit refactor: remove `cacheInfoTelemetry` altogether, just set `ApiEvent.CacheInfo` everywhere instead. (`ApiEvent.CacheInfo` is NotApplicable by default.)